include/exadg/operators/solution_projection_between_triangulations.cpp

```diff
@@ -0,0 +1,414 @@
+
+  // The sequence of integration points follows from the sequence of points as encountered during
+  // cell batch loop.
+  std::vector<dealii::Point<dim>> integration_points_target =
+    collect_integration_points<dim, n_components, Number>(target_matrix_free,
+                                                          dof_index,
+                                                          quad_index);
+
+  rpe_source.reinit(integration_points_target,
+                    source_dof_handler.get_triangulation(),
+                    *source_mapping);
+
+  if(not rpe_source.all_points_found())
+  {
+    write_points_in_dummy_triangulation(
+      integration_points_target, "./", "all_points", 0, source_dof_handler.get_communicator());
+
+    std::vector<dealii::Point<dim>> points_not_found;
+    points_not_found.reserve(integration_points_target.size());
+    for(unsigned int i = 0; i < integration_points_target.size(); ++i)
+    {
+      if(not rpe_source.point_found(i))
+      {
+        points_not_found.push_back(integration_points_target[i]);
+      }
+    }
+
+    write_points_in_dummy_triangulation(
+      points_not_found, "./", "points_not_found", 0, source_dof_handler.get_communicator());
+
+    AssertThrow(rpe_source.all_points_found(),
+                dealii::ExcMessage(
+                  "Could not interpolate source grid vector in target grid. "
+                  "Points exported to `./all_points.pvtu` and `./points_not_found.pvtu`"));
+  }
+
+  CellIntegrator<dim, n_components, Number> fe_eval(target_matrix_free, dof_index, quad_index);
+
+  // Loop over vectors and project.
+  for(unsigned int i = 0; i < target_vectors.size(); ++i)
+  {
+    // Evaluate the source vector at the target integration points.
+    VectorType const & source_vector = *source_vectors.at(i);
+    source_vector.update_ghost_values();
+
+    std::vector<
+      typename dealii::FEPointEvaluation<n_components, dim, dim, Number>::value_type> const
+      values_source_in_q_points_target = dealii::VectorTools::point_values<n_components>(
+        rpe_source, source_dof_handler, source_vector, dealii::VectorTools::EvaluationFlags::avg);
+
+    // Assemble right hand side vector for the projection.
+    VectorType system_rhs = assemble_projection_rhs<dim, n_components, Number, VectorType>(
+      target_matrix_free, fe_eval, values_source_in_q_points_target, dof_index);
+
+    // Solve linear system starting from zero initial guess.
+    VectorType sol;
+    sol.reinit(system_rhs, false /* omit_zeroing_entries */);
+
+    dealii::Timer              timer;
+    dealii::ConditionalOStream pcout(std::cout,
+                                     (dealii::Utilities::MPI::this_mpi_process(
+                                        source_dof_handler.get_communicator()) == 0));
+
+    timer.restart();
+
+    unsigned int n_iter = solver.solve(sol, system_rhs);
+
+    if(not data.is_test)
+    {
+      print_solver_info_linear(pcout, n_iter, timer.wall_time());
+    }
```

Comment:
-> `inverse_mass_operator.apply(dst, src)`

-> generic ExaDG infrastructure using the most efficient variant (e.g. explicit matrix-free inverse mass for DG + hypercube, elementwise inverse mass for DG + simplex, and Krylov solver for global problem otherwise (continuous Galerkin, RT))

Reply:
the reason I did not do this before was that I did not get that the `InverseMassOperatorHdiv` was essentially the most general case I needed. Now after the unification of the `InverseMassOperator`, I think we can mvoe forward with this in a nice way!